crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1672,6 +1690,80 @@ impl KnownInstanceType {
             _ => None,
         }
     }
+
+    fn member(self, db: &'db dyn Db, name: &str) -> Symbol<'db> {
+        match (self, name) {
+            (Self::TypeVar(typevar), "__name__") => Symbol::Type(
+                Type::StringLiteral(StringLiteralType::new(db, typevar.name(db).as_str())),
+                Boundness::Bound,
+            ),
+            (Self::TypeVar(typevar), "__bound__") => Symbol::Type(
+                typevar
+                    .upper_bound(db)
+                    .map(|ty| ty.to_meta_type(db))
+                    .unwrap_or(KnownClass::NoneType.to_instance(db)),
+                Boundness::Bound,
+            ),
+            (Self::TypeVar(typevar), "__constraints__") => Symbol::Type(
+                Type::Tuple(TupleType::new(
+                    db,
+                    typevar
+                        .constraints(db)
+                        .iter()
+                        .map(|ty| ty.to_meta_type(db))
+                        .collect::<Vec<_>>()
+                        .into_boxed_slice(),
+                )),
+                Boundness::Bound,
+            ),
+            (Self::TypeVar(typevar), "__default__") => Symbol::Type(
+                typevar
+                    .default_ty(db)
+                    .map(|ty| ty.to_meta_type(db))
+                    .unwrap_or(Type::KnownInstance(KnownInstanceType::NoDefault)),
+                Boundness::Bound,
+            ),
+            _ => self.instance_fallback(db).member(db, name),
+        }
+    }
+}
+
+#[salsa::interned]
+pub struct TypeVarInstance<'db> {
```

Comment:
I think the visibility of a lot of things in the `types` submodule is kind of chaotic right now 😆

Reply:
I left `bound_or_constraints` private because it should be accessed via the `upper_bound` and `constraints` methods instead. The discrepancy between `name` and `default_ty` was accidental.

I agree we may as well keep them all private until we need them to be otherwise; done!